### PR TITLE
Docs: Add Troubleshoot section

### LINKED ISSUE #1464: [Docs] Create a troubleshoot section
We have documented some issues in the getting started section, but there are some other issues users are running into, especially about installing `canvas`. We should document them and provide guidance.

Some example: https://github.com/Automattic/node-canvas/issues/1274 (no x86 version).

Original idea by @ingride

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.

## Short description of the change(s)

Fix #1464

### CODE REVIEW

**Comment by molant** [comment]
@ingride feel free to comment if there's anything else you think we should add!

File: packages/hint/docs/user-guide/troubleshoot/summary.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+binary is not available for download, the overall process should finish and
+you should be able to execute `webhint` using the `jsdom`. The only caveat
+is that images will not be downloaded.
+
+This error happens more often:
+
+* when there is a new release of node and precompiled binaries for `canvas`
+  are not yet available.
+* if you are running node x86 on Windows. The recommendation is to switch to
+  x64 as [it is unlikely there will be x85 binaries any time soon][canvas x86].
```
Comment (antross) [inline packages/hint/docs/user-guide/troubleshoot/summary.md]:
```suggestion
  x64 as [it is unlikely there will be x86 binaries any time soon][canvas x86].
```

File: packages/hint/docs/user-guide/troubleshoot/summary.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+npm install --global windows-build-tools
+```
+
+## Issues with `canvas`
+
+Starting on `connector-jsdom v1.1.0`, `canvas` was changed to be an optional
+dependency so while you might see some issues during the installation if the
+binary is not available for download, the overall process should finish and
+you should be able to execute `webhint` using the `jsdom`. The only caveat
```
Comment (antross) [inline packages/hint/docs/user-guide/troubleshoot/summary.md]:
```suggestion
you should be able to execute `webhint` using the `jsdom` connector. The only caveat
```

**Comment by molant** [comment]
Thanks @antross, I fixed those.

**Pull request merged**